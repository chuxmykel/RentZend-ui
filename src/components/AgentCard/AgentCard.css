.agent-card {
  display: flex;
  justify-content: space-around;
  align-items: center;
  min-height: 300px;
  width: 300px;
}

.agent-card-content {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
