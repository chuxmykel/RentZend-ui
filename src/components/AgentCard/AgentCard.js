import React from 'react';
import { Card, CardContent, Avatar, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import './AgentCard.css';

const useStyles = makeStyles(theme => ({
  large: {
    width: theme.spacing(15),
    height: theme.spacing(15),
  },
}));

const AgentCard = ({
  name,
  email,
  phoneNumber,
  address,
  zipCode,
  passport
}) => {
  const classes = useStyles();
  return (
    <Card className="agent-card" raised>
      <CardContent className="agent-card-content">
        <Avatar src={passport} alt={`agent - ${name}`} className={classes.large} />
        <Typography variant="h4">{name}</Typography>
        <Typography variant="subtitle1">{email}</Typography>
        <Typography variant="subtitle2">{phoneNumber}</Typography>
        <Typography variant="subtitle2">{address}</Typography>
        <Typography variant="subtitle2">{`zip code: ${zipCode}`}</Typography>
      </CardContent>
    </Card>
  )
}

export default AgentCard;
