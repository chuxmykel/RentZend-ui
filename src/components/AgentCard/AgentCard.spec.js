import React from 'react';
import { shallow } from 'enzyme';
import AgentCard from './AgentCard';
import data from '../../utils/dummy-data';

describe('<AgentCard />', () => {
  const container = shallow(<AgentCard {...data[0]} />);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
