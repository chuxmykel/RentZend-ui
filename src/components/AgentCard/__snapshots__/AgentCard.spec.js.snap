// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AgentCard /> should match the snapshot 1`] = `"<div class=\\"MuiPaper-root MuiCard-root agent-card MuiPaper-elevation8 MuiPaper-rounded\\"><div class=\\"MuiCardContent-root agent-card-content\\"><div class=\\"MuiAvatar-root MuiAvatar-circle makeStyles-large-1\\"><img alt=\\"agent - James Okorocha\\" src=\\"https://randomuser.me/api/portraits/men/75.jpg\\" class=\\"MuiAvatar-img\\"/></div><h4 class=\\"MuiTypography-root MuiTypography-h4\\">James Okorocha</h4><h6 class=\\"MuiTypography-root MuiTypography-subtitle1\\">james.o@gmail.com</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle2\\">(123) 456-7899</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle2\\">no. 10 Ojoro Street. Nakupenda</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle2\\">zip code: 101233</h6></div></div>"`;
